airbyte-integrations/connectors/source-shopify/unit_tests/test_transform.py

```diff
@@ -0,0 +1,137 @@
+                "buyer_accepts_marketing": {"type": ["null", "boolean"]},
+                "total_weight": {"type": ["null", "integer"]},
+                "created_at": {"type": ["null", "string"], "format": "date-time"},
+                "total_line_items_price": {"type": ["null", "number"]},
+                "id": {"type": ["null", "integer"]},
+                "total_tax": {"type": ["null", "number"]},
+                "subtotal_price": {"type": ["null", "number"]},
+                "line_items": {
+                    "items": {
+                        "properties": {
+                            "grams": {"type": ["null", "integer"]},
+                            "line_price": {"type": ["null", "number"]},
+                            "gift_card": {"type": ["null", "boolean"]},
+                            "price": {"type": ["null", "number"]},
+                        },
+                        "type": ["null", "object"],
+                    },
+                    "type": ["null", "array"],
+                },
+                "total_discounts": {"type": ["null", "number"]},
+                "note": {"type": ["null", "string"]},
+                "shipping_lines": {
+                    "items": {"properties": {"price": {"type": ["null", "number"]}}, "type": ["null", "object"]},
+                    "type": ["null", "array"],
+                },
+                "customer": {"type": "object", "properties": {"total_spent": {"type": ["null", "number"]}}},
+                "total_price": {"type": ["null", "number"]},
+            },
+            [
+                {
+                    "path": [
+                        "id",
+                    ],
+                    "expected_type": int,
+                },
+                {
+                    "path": [
+                        "buyer_accepts_marketing",
+                    ],
+                    "expected_type": bool,
+                },
+                {
+                    "path": [
+                        "note",
+                    ],
+                    "expected_type": type(None),
+                },
+                {
+                    "path": [
+                        "created_at",
+                    ],
+                    "expected_type": str,
+                },
+                {
+                    "path": [
+                        "total_discounts",
+                    ],
+                    "expected_type": Decimal,
+                },
+                {
+                    "path": [
+                        "total_price",
+                    ],
+                    "expected_type": Decimal,
+                },
+                {"path": ["customer", "total_spent"], "expected_type": Decimal},
+                {"path": ["line_items", 0, "line_price"], "expected_type": Decimal},
+                {"path": ["shipping_lines", 0, "price"], "expected_type": Decimal},
+            ],
+        )
+    ],
+)
+def test_transform(value, properties, checks):
```

Comment:
why don't we have `value, schema, expected`?
why do you need nested checks inside?
why `expected_type` and not actual value i.e. `10`?

Reply:
Shopify API returns price fields as a string. According to the issue them should be transformed to the number. I've changed schema. And according to it I transform values inside each records. Some records fields contain objects and arrays, which contain price fields. They should be transformed too. In this test I reproduce records with string price fields and schemes with required type, including fields with arrays and objects. And I check if price fields will be transformed to required type and other fields type will not change. That's why I used expected_type and not actual value and nested checks.